x-pack/plugins/observability_solution/logs_explorer/public/state_machines/logs_explorer_controller/src/services/data_view_service.ts

```diff
@@ -26,3 +26,14 @@ export const createAndSetDataView =
      */
     discoverStateContainer.actions.setDataView(dataView);
   };
+
+export const changeDataView =
+  (): InvokeCreator<LogsExplorerControllerContext, LogsExplorerControllerEvent> =>
+  async (context) => {
+    if (!('discoverStateContainer' in context)) return;
+    const { discoverStateContainer } = context;
+
+    return discoverStateContainer.actions.onChangeDataView(
```

Comment:
**note**: this action fetches and sets the complete data view object by its ID using the Discover pre-existing logic.